framework/src/org/checkerframework/framework/type/TypesIntoElements.java

```diff
@@ -244,7 +244,8 @@ private static void addUniqueTypeCompounds(Types types, Symbol sym, List<TypeCom
         List<Attribute.TypeCompound> res = List.nil();
 
         for (Attribute.TypeCompound tc : tcs) {
-            if (!TypeAnnotationUtils.isTypeCompoundContained(types, raw, tc)) {
+            // Don't insert an annotation if one with the same name already exists.
+            if (!TypeAnnotationUtils.isTypeCompoundContainedByName(types, raw, tc)) {
```

Comment:
My inclination would have been to canonicalize, too.  But I don't have a deep enough understanding of this code to make a strong pronouncement about it.